### PR TITLE
[10.0][ADD] partner_bank_sort_code: Add Sort Code on Banks

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
What is the goal of this? Do you mean `short code` instead?

**Comment by rousseldenis** [comment]
@pedrobaeza Nope. You didn't see this : https://github.com/OCA/partner-contact/pull/706/commits/ea6798937caf43b052c412c31d9981c69ddde1cb#diff-eec3b75ecd3331ec2ef5b65c5eb43154R30

**Comment by pedrobaeza** [comment]
Nope, I didn't see that. It seems a bit localization specific... but I don't block anything. Too much opened fences, hehe.

**Comment by rousseldenis** [comment]
That depends on countries, but as I know so far, UK, Ireland and Germany use these codes for interbanking transfers. That's the aim of this to modelize that feature.

File: partner_bank_sort_code/tests/test_sort_code.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        }
+        self.bank_2 = self.bank_obj.create(vals)
+
+    def test_sort_code_duplicate(self):
+        vals = {
+            'name': 'BANK 3',
+            'sort_code': '95-01-32',
+        }
+        with self.assertRaises(IntegrityError):
+            self.bank_obj.create(vals)
```
Comment (sbidoul) [inline partner_bank_sort_code/tests/test_sort_code.py:36]: add a test with active=False that should not raise?

Nice trick with the unique constraint.

**Review by sbidoul — APPROVED** [review]
A small test improvement suggestion. Otherwise LGTM.

File: partner_bank_sort_code/tests/test_sort_code.py (hunk 1 above)
Comment (rousseldenis) [reply partner_bank_sort_code/tests/test_sort_code.py:36]: @sbidoul Done

File: partner_bank_sort_code/models/res_bank.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+class ResBank(models.Model):
+
+    _inherit = 'res.bank'
+
+    sort_code = fields.Char()
+
+    _sql_constraints = [(
+        "sort_code_unique",
+        "EXCLUDE (sort_code WITH =) WHERE (active = True)",
```
Comment (lmignon) [inline partner_bank_sort_code/models/res_bank.py:16]: @rousseldenis  First time I read this syntax but it looks like you exclude the active record from the unique contrains. I've to check the postgresql documentation.

Comment (rousseldenis) [reply partner_bank_sort_code/models/res_bank.py:16]: Yes, the term is ambiguous. I've added tests and comments in order to be consistent

Comment (lmignon) [reply partner_bank_sort_code/models/res_bank.py:16]: @rousseldenis Great! Also tested on my side....

**Review by lmignon — APPROVED** [review]
LGTM (Code review)

**Review by Cedric-Pigeon — APPROVED** [review]
Code review :+1:

**Comment by rousseldenis** [comment]
this is green :smile:  :green_apple:

**Pull request merged**